### PR TITLE
Fix build dependency issue with settings and extension pages resources

### PR DESCRIPTION
**Pull request by petemill**
Resolves https://github.com/brave/brave-browser/issues/13567

This should be uplifted to wherever chromium 88 is

## Submitter Checklist:

- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed).
- [ ] Requested a security/privacy review as needed.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: patches/chrome-browser-resources-extensions-BUILD.gn.patch [hunk 1]
```diff
@@ -1,8 +1,16 @@
...
-index cfed148d68be8d0880823b391b55936f942c22cc..2b2989f4ce45b6b74d25d3537865d7d09c54c19e 100644
+index cfed148d68be8d0880823b391b55936f942c22cc..13eaf87e24b8203739cf5c51d4d0356f5052384e 100644
 --- a/chrome/browser/resources/extensions/BUILD.gn
 +++ b/chrome/browser/resources/extensions/BUILD.gn
-@@ -57,6 +57,7 @@ generate_grd("build_grd") {
+@@ -30,6 +30,7 @@ if (optimize_webui) {
+       "../../../../ui/webui/resources:preprocess",
+     ]
+     excludes = [ "chrome://resources/js/cr.m.js" ]
++    deps += brave_deps_chrome_browser_resources_extensions_build
```
Comment (bridiver) [inline patches/chrome-browser-resources-extensions-BUILD.gn.patch:9]: do we still need the other patch in this file? https://github.com/brave/brave-core/pull/7599/files#diff-5606dbee6795afa01ac32eebfe4d4cfc97f19d928fc34fac419a49599530a6adR17

**Review by bridiver — APPROVED** [review]
approving to unbreak the build, but I'd like to follow-up and make sure we have only the patches we need and not any extra

File: patches/chrome-browser-resources-extensions-BUILD.gn.patch (hunk 1 above)
Comment (bsclifton) [reply patches/chrome-browser-resources-extensions-BUILD.gn.patch:9]: I believe we do as this should be used for our settings customizations
cc: @petemill

Comment (petemill) [reply patches/chrome-browser-resources-extensions-BUILD.gn.patch:9]: @bridiver I believe we do as they are different sets of files used at different steps of the build

Comment (petemill) [reply patches/chrome-browser-resources-extensions-BUILD.gn.patch:9]: @bsclifton same applies to both settings and extensions. They both need 2 patches as far as I can tell so far.

**Comment by petemill** [comment]
re-building macos

**Review by bsclifton — APPROVED** [review]
++

**Review by mkarolin — APPROVED** [review]
++

**Comment by petemill** [comment]
Merging because CI is only being problematic for macOS and that's build-instance related, not PR-related. We've tested ourselves with macOS, all the other platforms have built ok with CI and this is only a dep addition aimed to fix CI.

**Pull request merged**